components/brave_shields/browser/ad_block_service.cc

```diff
@@ -218,95 +212,97 @@ absl::optional<base::Value> AdBlockService::HiddenClassIdSelectors(
 }
 
 AdBlockRegionalServiceManager* AdBlockService::regional_service_manager() {
-  if (!regional_service_manager_)
+  if (!regional_service_manager_) {
     regional_service_manager_ =
         brave_shields::AdBlockRegionalServiceManagerFactory(
-            component_delegate_);
+            local_state_, locale_, component_update_service_, GetTaskRunner());
+    regional_service_manager_->Init(resource_provider());
+  }
   return regional_service_manager_.get();
 }
 
-brave_shields::AdBlockCustomFiltersService*
-AdBlockService::custom_filters_service() {
-  if (!custom_filters_service_)
+brave_shields::AdBlockEngineService* AdBlockService::default_service() {
+  if (!default_service_) {
+    default_service_ = AdBlockEngineServiceFactory(GetTaskRunner());
+    default_service_->Init(default_source_provider_.get(),
+                           default_source_provider_.get());
+  }
+  return default_service_.get();
+}
+
+brave_shields::ResourceProvider* AdBlockService::resource_provider() {
+  return default_source_provider_.get();
+}
+
+brave_shields::AdBlockEngineService* AdBlockService::custom_filters_service() {
+  if (!custom_filters_service_) {
     custom_filters_service_ =
-        brave_shields::AdBlockCustomFiltersServiceFactory(component_delegate_);
+        brave_shields::AdBlockEngineServiceFactory(GetTaskRunner());
+    custom_filters_service_->Init(custom_filters_source_provider_.get(),
+                                  resource_provider());
+  }
   return custom_filters_service_.get();
 }
 
+brave_shields::AdBlockCustomFiltersSourceProvider*
+AdBlockService::custom_filters_source_provider() {
+  return custom_filters_source_provider_.get();
+}
+
 brave_shields::AdBlockSubscriptionServiceManager*
 AdBlockService::subscription_service_manager() {
+  if (!subscription_service_manager_->IsInitialized()) {
+    subscription_service_manager_->Init(resource_provider());
+  }
   return subscription_service_manager_.get();
 }
 
 AdBlockService::AdBlockService(
-    brave_component_updater::BraveComponent::Delegate* delegate,
+    PrefService* local_state,
+    std::string locale,
+    component_updater::ComponentUpdateService* cus,
+    scoped_refptr<base::SequencedTaskRunner> task_runner,
     std::unique_ptr<AdBlockSubscriptionServiceManager>
         subscription_service_manager)
-    : AdBlockBaseService(delegate),
-      component_delegate_(delegate),
-      subscription_service_manager_(std::move(subscription_service_manager)) {}
+    : local_state_(local_state),
+      locale_(locale),
+      component_update_service_(cus),
+      task_runner_(task_runner),
+      subscription_service_manager_(std::move(subscription_service_manager)) {
+  default_source_provider_ =
```

Comment:
the source provider should be owned by browser_process_impl and passed in to the AdBlockService

Reply:
well, it doesn't necessarily need to be owned by browser_process_impl right now, but it should be created in browser_process_impl and passed in